### ISSUE TITLE
[Feature Request] Provide support for Python > 3.7

### ISSUE THREAD

**Issue by jolespin** [ziyewang/COMEBin#29]
The following package could not be installed
└─ comebin is installable and it requires
   └─ python 3.7.* , which can be installed.

Adding support for more modern python versions will make this easier to integrate into new pipelines

**Comment by OH-AU** [comment]
I'd like to add my vote to this issue. With minor changes (change dtype np.int to just int ; change kmeans algorithm from full to lloyd, remove n_jobs=-1) I can get the training to complete in python/3.12.x however it fails at the point where the seed file is generated and then has issues with other scikit-learn related problems I think). We're running outside of any conda environment with local installs of all required software.

**Comment by jolespin** [comment]
I'd love to use this tool I just can't use it in my environment.

**Comment by jolespin** [comment]
Currently this tool is incompatible with CheckM2 so it can't be used in the same environment.

**Comment by hildebra** [comment]
Might I suggest that the community updates the public COMEbin repo on github to python 3.11? It's a bit of a shame that the developers have not found the time to take care of their own tool, especially such a simple and yet popular request.

**Comment by jolespin** [comment]
Other than containerized pipelines, it's hard to add it into other workflows

**Comment by hildebra** [comment]
Agreed, We wanted to integrate it into https://github.com/hildebra/mg-tk/ , but currently impossible. Will have to look at alternatives.

**Comment by ziyewang** [comment]
Thanks for the suggestion. COMEBin v1.1.0 now supports modern Python versions. The current Bioconda package supports Python 3.11–3.13, and the dependencies and installation instructions have been updated accordingly.
